src/api/worker/facades/LoginFacade.ts

```diff
@@ -405,6 +412,17 @@ export class LoginFacade {
 				return generateKeyFromPassphraseBcrypt(passphrase, salt, KeyLength.b128)
 			}
 			case KdfType.Argon2id: {
+				if (isIOSApp() || isAndroidApp()) {
```

Comment:
maybe inject some abstraction instead of doing wasm/native calls here?

Reply:
You're right. I moved all of the Argon2id stuff from LoginFacade to an Argon2idFacade in 88051545241c1a7f407339a8ff4bb05c22cc8160